contracts/GenericSwap.sol

```diff
@@ -53,41 +52,27 @@ contract GenericSwap is IGenericSwap, TokenCollector, EIP712 {
         if (!SignatureValidator.isValidSignature(taker, gs712Hash, takerSig)) revert InvalidSignature();
         filledSwap[swapHash] = true;
         returnAmount = _executeSwap(swapData, taker, takerTokenPermit);
-        emit Swap(
-            swapHash,
-            swapData.offer.maker,
-            swapData.offer.taker,
-            swapData.recipient,
-            swapData.offer.takerToken,
-            swapData.offer.takerTokenAmount,
-            swapData.offer.makerToken,
-            returnAmount
-        );
+        emit Swap(swapHash, swapData.maker, taker, swapData.recipient, swapData.takerToken, swapData.takerTokenAmount, swapData.makerToken, returnAmount);
     }
 
     function _executeSwap(
         GenericSwapData memory _swapData,
         address _authorizedUser,
```

Comment:
Maybe rename to taker?

Reply:
I think the `authorized` word is required so it's more clear for dev that this address has already been verified.

Reply:
I thought this address is not verified anymore?

Reply:
Either the address == `msg.sender`, or it's authorized by the taker signature. One cannot trade on other's behalf without proper authorization.